src/App.tsx

```diff
@@ -1,18 +1,50 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
+const items = getNumbers(1, 42).map((n) => `Item ${n}`);
+
+const perPageOptions: number[] = [3, 5, 10, 20];
 
 export const App: React.FC = () => {
+  const [current, setCurrent] = useState(1);
+  const [perPage, setPerPage] = useState(perPageOptions[0]);
+
+  const onPageChange = useCallback((page: number) => {
+    setCurrent(page);
+  }, []);
+
+  const onPerPageChange = useCallback(
+    (event: React.ChangeEvent<HTMLSelectElement>) => {
+      setPerPage(+event.target.value);
+      setCurrent(1);
+    },
+    [],
+  );
+
+  const lastVisibleItemIndex = useMemo(
+    () => current * perPage,
+    [current, perPage],
+  );
+
+  const firstVisibleItemIndex = useMemo(
+    () => lastVisibleItemIndex - perPage,
+    [lastVisibleItemIndex, perPage],
+  );
+
+  const itemsLength = useMemo(() => items.length, [items]);
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
```

Comment:
The displayed item indices should start from 1 instead of 0. Change the value of firstVisibleItemIndex to 'firstVisibleItemIndex + 1'.